.github/workflows/e2e.yml

```diff
@@ -0,0 +1,55 @@
+# Copyright (C) 2023, Ava Labs, Inc. All rights reserved.
+# See the file LICENSE for licensing terms.
+
+name: E2E Tests
+
+on:
+  push:
+    branches:
+      - main
+  pull_request:
+    branches:
+      - "*"
+
+env:
+  GO_VERSION: "1.20.7"
+
+jobs:
+  e2e_tests:
+    name: e2e_tests
+    runs-on: ubuntu-20.04
+
+    steps:
+      - name: Checkout repositories and submodules
+        uses: actions/checkout@v4
+        with:
+          submodules: recursive
+
+      - name: Setup Go
+        uses: actions/setup-go@v4
+        with:
+          go-version: ${{ env.GO_VERSION }}
+
+      - name: Install AvalancheGo Release
+        run: |
+          cd subnet-evm
+          BASEDIR=/tmp/e2e-test AVALANCHEGO_BUILD_PATH=/tmp/e2e-test/avalanchego ./scripts/install_avalanchego_release.sh
+
+      - name: Build Subnet-EVM Plugin Binary
+        run: |
+          cd subnet-evm
+          ./scripts/build.sh /tmp/e2e-test/avalanchego/plugins/srEXiWaHuhNyGwPUi444Tu47ZEDwxTWrbQiuD7FmgSAQ6X7Dy
+
+      - name: Checkout teleporter repository
+        uses: actions/checkout@v4
+
+      - name: Install Forge and Run E2E Tests
```

Comment:
It'd probably be slightly cleaner to split this into two steps, one for installing foundry, the other to run the test script.

Reply:
I tried that, but was running into issues persisting the `PATH` to include `foundryup` and `ginkgo`.

Reply:
I see this also consistent with our current workflow for running unit tests with `forge`, so it makes sense to keep it for now.